datafusion/core/src/catalog/information_schema.rs

```diff
@@ -81,15 +81,17 @@ impl InformationSchemaConfig {
 
             for schema_name in catalog.schema_names() {
                 if schema_name != INFORMATION_SCHEMA {
-                    let schema = catalog.schema(&schema_name).unwrap();
-                    for table_name in schema.table_names() {
-                        let table = schema.table(&table_name).await.unwrap();
-                        builder.add_table(
-                            &catalog_name,
-                            &schema_name,
-                            &table_name,
-                            table.table_type(),
-                        );
+                    // schema name may not exist in the catalog, so we need to check
+                    if let Some(schema) = catalog.schema(&schema_name) {
+                        for table_name in schema.table_names() {
+                            let table = schema.table(&table_name).await.unwrap();
+                            builder.add_table(
```

Comment:
What do you think about applying the same change to `table(..).unwrap()` here and the other places below? 

I think as written the code will still be subject to race conditions (if tables are removed from `Schema`s)

Reply:
Yes, it makes sense